apps/safe-claiming-app/src/widgets/ClaimingWidget.tsx

```diff
@@ -0,0 +1,209 @@
+import { CLAIMING_APP_URL } from "src/config/constants"
+import { useAirdropFile } from "src/hooks/useAirdropFile"
+import { useDelegate } from "src/hooks/useDelegate"
+import { useDelegatesFile } from "src/hooks/useDelegatesFile"
+import { useFetchVestingStatus } from "src/hooks/useFetchVestingStatus"
+import { useTokenBalance } from "src/hooks/useTokenBalance"
+import { sameAddress } from "src/utils/addresses"
+import { formatAmount } from "src/utils/format"
+
+const SpaceContent = styled("div")`
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  height: 300px;
+  padding: 24px;
+`
+
+const Title = (props: TypographyProps) => (
+  <Typography
+    color="primary.main"
+    style={{ fontWeight: "bold", textAlign: "center" }}
+  >
+    {props.children}
+  </Typography>
+)
+
+const Subtitle = (props: TypographyProps) => (
+  <Typography
+    variant="subtitle2"
+    color="primary.light"
+    style={{ marginBottom: "16px", textAlign: "center" }}
+  >
+    {props.children}
+  </Typography>
+)
+
+const StyledButton = (props: ButtonProps) => (
+  <Button
+    size="large"
+    onClick={props.onClick}
+    variant="contained"
+    disableElevation
+    fullWidth
+    disabled={props.disabled}
+  >
+    {props.children}
+  </Button>
+)
+
+const WIDGET_HEIGHT = 300
+
+const ClaimingWidget = () => {
+  const [delegates] = useDelegatesFile()
+  const delegateAddressFromContract = useDelegate()
+  const [balance, , balanceLoading] = useTokenBalance()
+  const { safe } = useSafeAppsSDK()
+
+  const [vestings] = useAirdropFile()
+  const [userVesting, ecosystemVesting, investorVesting] = vestings
+
+  const totalAllocation = BigNumber.from(userVesting?.amount || 0)
+    .add(ecosystemVesting?.amount || 0)
+    .add(investorVesting?.amount || 0)
+
+  const currentDelegate = useMemo(() => {
+    if (delegateAddressFromContract) {
+      const registeredDelegateFromData = delegates.find((entry) =>
+        sameAddress(entry.address, delegateAddressFromContract)
+      )
+      return (
+        registeredDelegateFromData || { address: delegateAddressFromContract }
+      )
+    }
+  }, [delegateAddressFromContract, delegates])
+
+  const [userVestingStatus, , userVestingLoading] = useFetchVestingStatus(
+    userVesting?.vestingId,
+    userVesting?.contract
+  )
+
+  const [ecosystemVestingStatus, , ecosystemVestingLoading] =
+    useFetchVestingStatus(
+      ecosystemVesting?.vestingId,
+      ecosystemVesting?.contract
+    )
+
+  const [investorVestingStatus, , investorVestingLoading] =
+    useFetchVestingStatus(investorVesting?.vestingId, investorVesting?.contract)
+
+  const totalClaimed = BigNumber.from(userVestingStatus?.amountClaimed || 0)
+    .add(ecosystemVestingStatus?.amountClaimed || 0)
+    .add(investorVestingStatus?.amountClaimed || 0)
+
+  const votingPower = useMemo(() => {
+    if (totalAllocation.gt(0) && totalClaimed.gt(0) && balance?.gt(0)) {
```

Comment:
People can have voting power without claiming. For instance investors can use the vested tokens to vote without ever claiming any tokens. Meaning totalClaimed and balance will be 0.

Reply:
We should use the previous math here: `totalAllocation + balance - totalClaimed` why did we change it?

Reply:
I changed it while implementing the loading Skeleton but didn't realize that mistake. I will add it back 
```js
  const votingPower = totalAllocation
    .add(balance || BigNumber.from(0))
    .sub(totalClaimed)